### PR TITLE
[APPS-510] Offline video MediaRecorder class

### PR DESCRIPTION
**Pull request by katelyn-gallagher**
[APPS-510](https://blinemedical.atlassian.net/browse/APPS-510)

## Overview
- Implement typescript class for MediaRecorder
- Functions don't need to do anything yet

## Changes
- Created class, and included a bunch of research on codecs/mime-types in a comment
- Created constructor and functions requested in the ticket, but they don't do anything yet
- Setup sym-links, but still need to test with SCMR to see if they're fully working

## Testing
- Nothing to test yet

[APPS-510]: https://blinemedical.atlassian.net/browse/APPS-510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Gandri — CHANGES_REQUESTED** [review]
Super small feedback nothing crazy :)

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (katelyn-gallagher) [reply .vscode/settings.json:1]: Lol no, I did not. Oops :)

File: src/MediaRecorder.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    }
+
+    /**
+     * On 'stop' event handler; 'stop' is fired when the recording stops
+     */
+    onstop = (event: Event) => {
+        // TODO: save the file to the filePath specified when we started recording
+        throw Error('Unimplemented');
+    };
```
Comment (Gandri) [inline src/MediaRecorder.ts:80]: Per our discussion yesterday, we should follow the conventions of the library for this. Which will involve additional effort. We can for sure kick that off to a separate ticket though! :)  If that's the case let's remove this for now and add a comment to the class saying we will implement this event later

**Review by Gandri — APPROVED** [review]
Looks much better. Yeah I think locking things down for now is fine, if we wanted we could expose mimeType as a constructor arg so we could override it but likely we can easily change that later when we are ready to experiment more

**Pull request merged**